client/modules/wj-state/src/useragent.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Browser / User-Agent info. Contains contextual information like
+ * normalized mouse position values.
+ */
+export namespace UserAgent {
+  /** Horizontal mouse position, normalized between 0-1. */
+  export let mouseX = 0
+  /** Vertical mouse position, normalized between 0-1. */
+  export let mouseY = 0
+  /** Scroll position, normalized between 0-1. */
+  export let scroll = 0
+  /**
+   * Flag that is true if the agent is using a mobile device. Probably,
+   * anyways. It's difficult to rely on this and it should only be used
+   * with caution.
+   */
+  export const isMobile = /Mobi|Android/i.test(navigator.userAgent)
+
+  // set up our listeners
+
+  window.addEventListener("mousemove", evt => {
+    mouseX = evt.clientX / window.innerWidth
+    mouseY = evt.clientY / window.innerHeight
+  })
```

Comment:
??????

why??????

Reply:
you won't need these values for precise location, instead it's for rough location with no effort involved.

e.g. if you want a navbar to pop down if the mouse is near the top of the screen, you can just check if mouseY is less than 0.10 or something like that. otherwise you'd have to kinda guestimate a pixel range, which isn't reliable for different input device size

Reply:
Huh, alright. I'm down with that.